sdk/core/src/error/azure_core_errors.rs

```diff
@@ -19,6 +19,10 @@ impl From<crate::errors::Error> for Error {
             crate::errors::Error::GetToken(e) => Error::new(ErrorKind::Credential, e),
             crate::errors::Error::HttpPrepare(e) => e.into(),
             crate::errors::Error::Other(e) => Error::new(ErrorKind::Other, e),
+            crate::errors::Error::AuthorizationPolicy(msg) => Error::with_message(
```

Comment:
If `AuthorizationPolicy` takes a boxed error trait object, you can just use `Error::new` instead of `Error::with_message` which doesn't really add much value.

Reply:
It seems that when working with the guard from RwLock, we need to get a reference (`guard.as_ref()`) , which in turn ties the lifetime of the error to the TokenCredential itself, which then would need to be constrained by the 'static lifetime.

At least that's more or less what i gathered from the compiler errors, I get when trying to pass a box'ed error into the variant. Tried around different variations of de-referencing the error, but with no success. Somehow these lifetimes still manage to escape my grasp every so often :).